app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/DraftController.kt

```diff
@@ -188,27 +191,72 @@ class DraftController @Inject constructor(
             return@let document.outerHtml()
         } ?: ""
 
-        val ccList = if (message.cc.isNotEmpty()) {
-            "<div>$ccTitle ${message.cc.joinToString { it.quotedDisplay() }}<br></div>"
-        } else {
-            ""
+        val previousFullBody = computePreviousFullBody(previousBody, message)
+
+        return assembleForwardHtmlHeader(
+            messageForwardHeader,
+            fromTitle,
+            message,
+            dateTitle,
+            subjectTitle,
+            toTitle,
+            ccTitle,
+            previousFullBody
+        )
+    }
+
+    private fun assembleForwardHtmlHeader(
+        messageForwardHeader: String,
+        fromTitle: String,
+        message: Message,
+        dateTitle: String,
+        subjectTitle: String,
+        toTitle: String,
+        ccTitle: String,
+        previousFullBody: String
+    ): String {
+        val forwardRoot = "<div class=\"${MessageBodyUtils.INFOMANIAK_FORWARD_QUOTE_HTML_CLASS_NAME}\" />"
+        return newDocumentWithParentElement(forwardRoot).apply {
+            addAndEscapeTextLine("---------- $messageForwardHeader ---------")
+            addAndEscapeTextLine("$fromTitle ${message.fromName()}")
+            addAndEscapeTextLine("$dateTitle ${message.date.toDate()}")
+            addAndEscapeTextLine("$subjectTitle ${message.subject}")
+            addAndEscapeRecipientLine(toTitle, message.to)
+            addAndEscapeRecipientLine(ccTitle, message.cc)
+            addAndEscapeTextLine("")
+            addAndEscapeTextLine("")
+
+            addAlreadyEscapedBody(previousFullBody)
```

Comment:
Is it normal that we add the body in a function assembling an header ?

Reply:
Maybe naming it "footer" would be more fitting as this is actually what comes at the end of emails